src/transformers/generation/utils.py

```diff
@@ -4507,6 +4507,8 @@ def assisted_decoding(
         if streamer is not None:
             streamer.end()
 
+        if candidate_generator.assistant_model.generation_config.num_assistant_tokens_schedule == "heuristic":
```

Comment:
We need to check first that `candidate_generator.assistant_model` exists -- ngram speculation doesn't have this attribute, which is causing its tests to fail

Reply:
sure!
also relevant for speculation with PLD